src/runtime.js

```diff
@@ -417,9 +421,14 @@ export function adopt(global) {
  * @param {!Window} global Global scope to adopt.
  */
 export function adoptShadowMode(global) {
+
+  const doc = global.document,
+    isCriticalPath = !!doc.querySelector('script[shadow-doc-for-critical-path]');
```

Comment:
Our concept of "critical path" is equivalent to your `AmpDocShell`: also create and AmpDoc for `window.document`when AMP runtime is in shadow mode.

We did this specific change because we though that having an `AmpDocShell` could be optional. Perhaps someone wanted to use AMP Shadow without AMP managing the shell.

Do you think it's safe to always instantiate `AmpDocShell`?